src/App.tsx

```diff
@@ -1,12 +1,74 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
 import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import cN from 'classnames';
+
+type Filter = 'All' | 'Completed' | 'Active';
+
+function filterTodos(todos: Todo[], completedStatus: boolean) {
+  return todos.filter(todo => todo.completed === completedStatus);
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [isAllCompleted, setIsAllCompleted] = useState(false);
+  const [hasCompleted, setHasCompleted] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [activeFilter, setActiveFilter] = useState<Filter>('All');
+  const [activeCount, setActiveCount] = useState<number>(0);
+  const addTodoField = useRef<HTMLInputElement>(null);
+
+  function ShowError(message: string) {
+    setErrorMessage(message);
+    setTimeout(() => setErrorMessage(null), 3000);
+  }
+
+  useEffect(() => {
+    if (todos.length !== 0) {
+      setIsAllCompleted(!todos.some(todo => todo.completed === false));
+      setHasCompleted(todos.some(todo => todo.completed === true));
+      setActiveCount(
+        todos.reduce((acc, todo) => {
+          if (todo.completed === false) {
+            return acc + 1;
+          }
+
+          return acc;
+        }, 0),
+      );
+    }
+  }, [todos]);
```

Comment:
Try to simplify your solution
Use just variables instead of `useEffect` and states that I wrote above